server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -317,10 +322,15 @@ public void onResponse(SearchResponse searchResponse) {
                             }
                         }
                     }
-                    listener.onResponse(searchResponse);
                     // increment after the delegated onResponse to ensure we don't
                     // record both a success and a failure if there is an exception
                     searchResponseMetrics.incrementResponseCount(responseCountTotalStatus);
+
+                    if (CCS_TELEMETRY_FEATURE_FLAG.isEnabled() && searchResponse.getClusters().hasRemoteClusters()) {
+                        extractCCSTelemetry(searchResponse, (SearchTask) task);
+                    }
+                    // TODO: should this be last?
+                    listener.onResponse(searchResponse);
```

Comment:
We're doing things like counting searches here, but only for successful searches. Shouldn't we count both successful and failed ones?